pkg/api/agentPoolOnlyApi/v20170831/types.go

```diff
@@ -94,6 +94,26 @@ const (
 	Migrating ProvisioningState = "Migrating"
 )
 
+// PoolUpgradeProfile contains pool properties:
+//  - kubernetes version
+//  - pool name (for agent pool)
+//  - OS type of the VMs in the pool
+//  - list of applicable upgrades
+type PoolUpgradeProfile struct {
```

Comment:
Can we name it AgentPoolUpgradeProfile instead?

Reply:
Nvm. Looks like it's used by ControlPlane as well as by AgentPool

Reply:
It also works for master/controlPlane pool